### DISCUSSION TITLE
How to ssh into aws ec2

### DISCUSSION THREAD

**Discussion by vivianc2** [CornellDataScience/MathSearch#39] [Q&A] [answered]
How to ssh into aws ec2？

**Comment by emerisly** [comment, marked as answer]
## 1. Just ssh (easy but less convenient for long-term usage)
Ask Emerald, Felix, or Zander for the certificate file "mathsearch-backend.pem".
Do not give this file to others or upload it GitHub.
Put pem file under the dir you want to work in, cd into that dir.
Change pem file security by running ```chmod 600 mathsearch-backend.pem```.
Now we are ready to ssh into ec2, run ```ssh -i "mathsearch-backend.pem" ubuntu@ec2-18-207-249-45.compute-1.amazonaws.com```.


## 2. Recommend method with vscode
cd into .ssh dir with ```cd ~/.ssh``` command.
Put pem file under this dir and run ```chmod 600 mathsearch-backend.pem```.
Edit file ~/.ssh/config by adding below few lines. This adds ssh permission so your vscode can recognize.
```
# aws
Host ec2-18-207-249-45.compute-1.amazonaws.com
   HostName ec2-18-207-249-45.compute-1.amazonaws.com
   User ubuntu
   IdentityFile ~/.ssh/mathsearch-backend.pem
```
Open vscode, make sure you have remote-ssh extension, do "Remote SSH: Connect to Host" and select our was.
On the left, you can select folder path to show in explorer.
